### ISSUE TITLE
(web) (html-renderer) implement toImage

### ISSUE THREAD

**Issue by bw-flagship** [flutter/flutter#102636]
## Use case

It's about the method toImage():
https://api.flutter.dev/flutter/rendering/OffsetLayer/toImage.html

It is extremely useful for many scenarios, and for example used in the famous screenshot-package:
https://pub.dev/packages/screenshot

It works on Android and iOS. Furthermore, it works on web as well, if you use the canvaskit-renderer.
However, there are scenarios where the html-render is required to avoid other missing features.
Therefore, it would be a great improvement if the toImage-Method would work with the html renderer as well.

## Proposal
I propose to implement the toImage method in the html renderer. This cannot be replaced by pub-packages.

## Proposal light
If this is not possible for any reason, a small but good step would be to update the error message. Currently, it says, "toImage is not supported on the web", but in reality it is supported if you use the canvaskit.

![2022-04-27_12-02-23](https://user-images.githubusercontent.com/89379862/165494919-9c1de0ce-ed49-4051-8551-0357aece1d79.png)

## Some code to reproduce
If you want to reproduce the fact that it works on canvaskit but not on html, you can use the following code.

```
// ignore_for_file: avoid_print

import 'dart:typed_data';

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';
import 'dart:ui' as ui;

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Demo',
      home: PngHome(),
    );
  }
}

class PngHome extends StatefulWidget {
  const PngHome({Key? key}) : super(key: key);

  @override
  State<PngHome> createState() => _PngHomeState();
}

class _PngHomeState extends State<PngHome> {
  GlobalKey globalKey = GlobalKey();

  Future<void> _capturePng() async {
    final RenderRepaintBoundary boundary =
        globalKey.currentContext!.findRenderObject()! as RenderRepaintBoundary;
    final ui.Image image = await boundary.toImage();
    final ByteData? byteData =
        await image.toByteData(format: ui.ImageByteFormat.png);
    final Uint8List pngBytes = byteData!.buffer.asUint8List();
    print(pngBytes);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: RepaintBoundary(
        key: globalKey,
        child: Center(
          child: ElevatedButton(
            child: const Text('Hello World', textDirection: TextDirection.ltr),
            onPressed: _capturePng,
          ),
        ),
      ),
    );
  }
}

```

Run it via
`flutter run -d chrome --web-renderer html`
and via
`flutter run -d chrome --web-renderer canvaskit`

You will see in the console that the html-thing throws an error, while the canvaskit works.

**Comment by darshankawar** [comment]
@bw-flagship 
Thanks for the report.



> If this is not possible for any reason, a small but good step would be to update the error message. Currently, it says, "toImage is not supported on the web", but in reality it is supported if you use the canvaskit.

I think this open issue will likely address this: https://github.com/flutter/flutter/issues/49857. Also see [this](https://github.com/flutter/flutter/issues/49857#issuecomment-586020711). 

I am closing this as duplicate of linked issue. If you disagree, write in comments and I'll reopen it.

**Comment by bw-flagship** [comment]
@darshankawar improving the docs was just a sidenote of the issue, the main thing was about enabling toImage for html. However, I found another one which really seems to be a duplicate:
#59072

**Comment by darshankawar** [comment]
Yes, thanks for the feedback. I had gone through that issue too, but thought it was specific to `toByte()` method.
